### PR TITLE
Test VSIX release for releasebranch

### PR DESCRIPTION
**Pull request by ebarboni**
would just try to put that in place to test for release vsix alongisde main release. added param for vsix version and some ant target and copy.

scanner will not work on vs* branches now

### CODE REVIEW

**Review by JaroslavTulach — COMMENTED** [review]
I don't understand the change. There is no call to `ant -f java/java.lsp.server test-vscode-ext` as far as I can tell. So I don't understand why the title says _"Test VSIX release for releasebranch"_.

File: vars/asfMainNetBeansBuild.groovy [hunk 1]
```diff
@@ -273,111 +281,115 @@ def doParallelClusters(cconfigs) {
...
                                     sh "mvn org.apache.netbeans.utilities:nb-repository-plugin:${repopluginversion}:populate ${commonparam} -DnetbeansNbmDirectory=${netbeansbase}/nbms -DnetbeansInstallDirectory=${netbeansbase}/netbeans -DnetbeansSourcesDirectory=${netbeansbase}/build/source-zips -DnetbeansJavadocDirectory=${netbeansbase}/build/javadoc -DparentGAV=org.apache.netbeans:netbeans-parent:2 -DforcedVersion=${mavenVersion} -DskipInstall=true -DdeployUrl=file://${env.WORKSPACE}/mavenrepository"
-                                }                            
+                                }
                                 archiveArtifacts 'mavenrepository/**'
-                                                   
+
+                                // make vsix available to dist to pickup (only for main release) 
+                                sh "ant -f build-${clustername}-temp/java/java.lsp.server build-vscode-ext -Dvsix.version=${vsixversion}"
+                                sh "cp -r build-${clustername}-temp/java/java.lsp.server/build/*.vsix dist/vsix/"
```
Comment (ebarboni) [inline vars/asfMainNetBeansBuild.groovy:376]: Sorry @JaroslavTulach the line of release for vsix are there. I didn't notice there were so many trailing space.

Comment (ebarboni) [reply vars/asfMainNetBeansBuild.groovy:376]: And sorry again I would test how it works to build the vsix as we do for exe so not do the junit test

**Review by JaroslavTulach — APPROVED** [review]
As far as I can see, the PR builds `.vsix` file via `ant -f java/java.lsp.server build-vscode-ext` and archives it. That's great. Thank you for including `.vsix` into the official list of checked artifacts.

**Comment by neilcsmith-net** [comment]
Seems a good idea.  As I guess we're sticking with a single json file :shrug: ... let's make it work for us.

I guess other than adding flags for whether it's a VSNetBeans and/or IDE release, we could also use presence of these version fields?  Would mean making sure anything just for the IDE checks for a setting for `versionName`.

**Review by neilcsmith-net — APPROVED** [review]

**Comment by ebarboni** [comment]
as build 4 is ready and  locked I merge this to check if it's ok

**Comment by ebarboni** [comment]
@JaroslavTulach first vsix bits present here:
https://ci-builds.apache.org/job/Netbeans/job/netbeans-TLP/job/netbeans/job/release130/lastSuccessfulBuild/artifact/dist/vsix/

Would be nice to know if it's ok so far. bundling and so one.

**Comment by JaroslavTulach** [comment]
I have downloaded the `.vsix`, installed it and it does seem to run sufficiently well. Change log isn't updated for 13 and Martin Balín reported other misbehavior in new functionality, like:
```
Warning - could not install some modules:
	org.netbeans.modules.cloud.oracle - The module named bcprov was needed and not found.
	org.netbeans.modules.cloud.oracle - The module named com.google.guava was needed and not found.
	org.netbeans.modules.cloud.oracle - The module named org.netbeans.modules.websvc.restlib/2 was needed and not found.
	org.netbeans.modules.cloud.oracle - The module named org.apache.commons.httpclient was needed and not found.
```
but that is problem somewhere else. From a release point of view, it is probably good enough complementary binary.

**Comment by ebarboni** [comment]
ok will work later on the previous branches :p will be fun ;p

**Comment by neilcsmith-net** [comment]
@ebarboni unfortunately rc2 build failed in lsp build (9).  Failure to execute mvn.  Retrying (10).  Will also check build.xml changes in sources not the cause.  If this, we may need to revert for now?

**Comment by ebarboni** [comment]
I've moved the ant call under a withmaven directive. I restarted the build. Hope it will be ok

**Comment by neilcsmith-net** [comment]
OK. You aborted 10?  I was curious it seemed to build 7 fine.  I've locked that build for now in case we need to review log, etc.

**Comment by ebarboni** [comment]
maybe the node selection were lucky and has mvn built in

**Comment by neilcsmith-net** [comment]
Looks like that was the key, thanks.  Build 11 has completed successfully so will check and announce asap.  Build 7 was definitely on a different node - looked at a few things, but hadn't considered the "lucky" availability of mvn on there.

**Pull request merged**